extensions/positron-r/src/provider.ts

```diff
@@ -258,198 +237,101 @@ export async function makeMetadata(
 	return metadata;
 }
 
-// directory(ies) where this OS is known to keep its R installations
-function rHeadquarters(): string[] {
-	switch (process.platform) {
-		case 'darwin':
-			return [path.join('/Library', 'Frameworks', 'R.framework', 'Versions')];
-		case 'linux':
-			return [path.join('/opt', 'R')];
-		case 'win32': {
-			const paths = [
-				path.join(process.env['ProgramW6432'] || 'C:\\Program Files', 'R')
-			];
-			if (process.env['LOCALAPPDATA']) {
-				paths.push(path.join(process.env['LOCALAPPDATA'], 'Programs', 'R'));
-			}
-			return [...new Set(paths)];
-		}
-		default:
-			throw new Error('Unsupported platform');
-	}
-}
+// functions relating to the current R binary
```

Comment:
From here down, there is a mix of actual changes, which I'll comment on, and simply reorganizing a file that had become super messy.